catalog/core/catalog-core-definitionparser/src/main/java/ddf/catalog/definition/impl/DefinitionParser.java

```diff
@@ -316,6 +354,48 @@ private void commitStaged(List<Callable<Boolean>> stagedAdds) throws Exception {
     return staged;
   }
 
+  private void processAttribute(
+      Outer.MetacardType metacardType,
+      /*Mutable*/ Set<AttributeDescriptor> attributeDescriptors,
+      /*Mutable*/ Set<String> requiredAttributes,
+      String attributeName,
+      ddf.catalog.definition.impl.DefinitionParser.Outer.MetacardAttribute attribute) {
+    AttributeDescriptor descriptor =
+        attributeRegistry
+            .lookup(attributeName)
+            .orElseThrow(
+                () ->
+                    new IllegalStateException(
+                        String.format(
+                            "Metacard type [%s] includes the attribute [%s], but that attribute is not in the attribute registry.",
+                            metacardType.type, attributeName)));
+    attributeDescriptors.add(descriptor);
+    if (attribute.required) {
+      requiredAttributes.add(attributeName);
+    }
+  }
+
+  private Function<String, Stream<? extends AttributeDescriptor>> getSpecifiedTypes(
+      List<MetacardType> stagedTypes) {
+    return type ->
+        new ImmutableList.Builder<MetacardType>()
+            .addAll(metacardTypes)
+            .addAll(stagedTypes)
+            .addAll(coreTypes)
+            .build()
+            .stream()
+            .filter(mt -> mt.getName().equals(type))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new RuntimeException(
+                        String.format(
+                            "Could not find a metacard type by name '%s'. Was the type already defined or defined first in the list of definitions?",
```

Comment:
Above, you wrapped `%s` with square brackets, but here you use single quotes. Are you OK with using the same quoting style in all of the exceptions?

Reply:
I went ahead and fixed all the instances in this file to use consistent string format of single quotes. Figured single quotes makes more sense because i'd use hard brackets if it were a list of values.